indexer/services/comlink/src/controllers/api/v4/historical-pnl-controller.ts

```diff
@@ -86,6 +86,84 @@ class HistoricalPnlController extends Controller {
       }),
     };
   }
+
+  @Get('/parentSubaccount')
+  async getHistoricalPnlForParentSubaccount(
+    @Query() address: string,
+      @Query() parentSubaccountNumber: number,
+      @Query() limit?: number,
+      @Query() createdBeforeOrAtHeight?: number,
+      @Query() createdBeforeOrAt?: IsoString,
+      @Query() createdOnOrAfterHeight?: number,
+      @Query() createdOnOrAfter?: IsoString,
+  ): Promise<HistoricalPnlResponse> {
+
+    const childSubaccountIds: string[] = getChildSubaccountIds(address, parentSubaccountNumber);
+
+    const [subaccounts, pnlTicks]: [
+      SubaccountFromDatabase[],
+      PnlTicksFromDatabase[],
+    ] = await Promise.all([
+      SubaccountTable.findAll(
+        {
+          id: childSubaccountIds,
+        },
+        [QueryableField.ID],
+      ),
+      PnlTicksTable.findAll(
+        {
+          subaccountId: childSubaccountIds,
+          limit,
+          createdBeforeOrAtBlockHeight: createdBeforeOrAtHeight
+            ? createdBeforeOrAtHeight.toString()
+            : undefined,
+          createdBeforeOrAt,
+          createdOnOrAfterBlockHeight: createdOnOrAfterHeight
+            ? createdOnOrAfterHeight.toString()
+            : undefined,
+          createdOnOrAfter,
+        },
+        [QueryableField.LIMIT],
+        {
+          ...DEFAULT_POSTGRES_OPTIONS,
+          orderBy: [[QueryableField.BLOCK_HEIGHT, Ordering.DESC]],
+        },
+      ),
```

Comment:
don't you want the to pull both the childSubaccountIds and the parentSubaccountId? You're ignoring the parent now

Reply:
The childSubccountIds are inclusive. So parentSubaccountNumber 0 will have childSubaccountNumbers 0, 128, 256 , ...

Reply:
It does? https://github.com/dydxprotocol/v4-chain/blob/0dbaf0b0df4a684d58233acc275a79e6988688ff/indexer/services/comlink/src/lib/helpers.ts#L504
Is there anything more you'd like to have specified?